### ISSUE TITLE
Error when debugging a non-local target

### ISSUE THREAD

**Issue by BigFatFlo** [cyrus-and/gdb-dashboard#207]
Hi,

I'm using (and loving) the dashboard without any issues on a local target, connecting through OpenOCD running locally (accessed through localhost:1234).
But I'm having trouble using the dashboard when I'm debugging a target that is not accessible through a local instance of OpenOCD. When I'm running OpenOCD on a remote machine (accessed through remote_machine:1234) and I launch gdb I get this error:
```
Cannot write the dashboard
Traceback (most recent call last):
  File "<string>", line 386, in render
  File "<string>", line 1385, in lines
AttributeError: 'NoneType' object has no attribute 'is_stopped'`
/gdb/thread.c:100: internal-error: thread_info* inferior_thread(): Assertion `tp' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Quit this debugging session? (y or n) 
```
Only the target location is different from my usual debug process.
Please note that if I disable the dashboard, I'm able to use gdb normally.
However, if I try to change the layout, for instance with 'ctrl + X; 2', it also crashes my debug session.

Any idea what might cause this issue? Could it be a timeout issue somewhere? The commands do take a longer than usual time to execute given the remote connection.

Thank you

**Comment by cyrus-and** [comment]
Since I know nothing about OpenOCD, how can I try to reproduce this issue on my end? Is OpenOCD required to reproduce the issue? Will any other remote usage do as well?

What if you try (without the dashboard; when the program is paused) to run the following?

```
(gdb) python print(gdb.selected_thread())
<gdb.InferiorThread object at 0x7f49e9500a10>
```

This part hints that this might be a GDB bug, even though this issue is triggered by the dashboard it shouldn't be possible to violate code assertions.

```
/gdb/thread.c:100: internal-error: thread_info* inferior_thread(): Assertion `tp' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
```

> However, if I try to change the layout, for instance with 'ctrl + X; 2', it also crashes my debug session.

This is TUI stuff which is not compatible with the dashboard anyway.

**Comment by BigFatFlo** [comment]
Thank you for your quick reply!
I'm happy to report that my issue has been fixed and that it had nothing to do with the dashboard after all.
Turns out that the version of OpenOCD that was installed on the remote server wasn't the same as the local one and evidently had some compatibility issues with the gdb I'm using that only manifested when I used the dashboard.
Sorry for the inconvenience, I'll close the issue.
